test/Graphics/Text.test.cpp

```diff
@@ -182,4 +185,47 @@ TEST_CASE("[Graphics] sf::Text", runDisplayTests())
             CHECK(text.getGlobalBounds() == Approx(sf::FloatRect({66, 182}, {33, 13})));
         }
     }
+
+#ifdef SFML_ENABLE_LIFETIME_TRACKING
+    SECTION("Lifetime tracking")
+    {
+        SECTION("Return local from function")
+        {
+            const auto badFunction = []
+            {
+                const auto localFont = sf::Font::loadFromFile("Graphics/tuffy.ttf").value();
+                return sf::Text(localFont);
+            };
+
+            const sf::priv::LifetimeDependee::TestingModeGuard guard;
```

Comment:
The tests currently make no mention of the `priv::` namespace and I'd like to keep it that way. There are definitely some `sf::priv::` things I wish we could test but I'm rather dogmatic about the unit tests only using the public API and am not comfortable testing implementation details like this which users are not intended to use or even know about.

How much code could be removed from this PR if `TestingModeGuard` was removed?

This code will still be tested although instead of explicitly testing for failure we will be implicitly testing for the lack of failure. I know that's not the same but by simply using types like `sf::Sprite` we will still be testing that your lifetime tracking has no false positives. We cannot test for true negatives quite as well though but I think I'm okay with that. It's the same as all of our `assert`s. We can't test for assertion failure but we can test that assertions don't have false positives. In practice this seems to work quite well.